ee/clickhouse/queries/experiments/funnel_experiment_result.py

```diff
@@ -292,3 +296,27 @@ def calculate_probability_of_winning_for_each(variants: List[Variant]) -> List[P
     total_test_probabilities = sum(probabilities[1:])
 
     return [max(0, 1 - total_test_probabilities), *probabilities[1:]]
+
+
+def validate_event_variants(filtered_results, variants):
+    if not filtered_results or not filtered_results[0]:
+        raise ValidationError("No experiment events have been ingested yet")
```

Comment:
I'd _love_ to distinguish between the case where events are coming in without flag information, and where no events have been ingested, because they lead to very different user messages:

(1) is almost always implementation error / some action that the user needs to take (like updating code, or updating experiment metric)
(2) is almost always 'wait longer'

I think if you validate before filtering, we'll get this distinction.